File: src/main/java/org/hotamachisubaru/miniutility/Nickname/NicknameConfig.java
package org.hotamachisubaru.miniutility.Nickname;

import org.bukkit.entity.Player;

import java.sql.SQLException;
import java.util.HashMap;
import java.util.Map;
import java.util.UUID;

public class NicknameConfig {
    private final Map<UUID, Boolean> waitingForNicknameInput = new HashMap<>();

    public void setNickname(UUID uuid, String nickname) throws SQLException {
        NicknameDatabase.saveNicknameToDatabase(uuid.toString(), nickname);
    }

    public String getNickname(UUID uuid, String defaultName) throws SQLException {
        String nickname = NicknameDatabase.loadNicknameFromDatabase(uuid.toString());
        return (nickname != null) ? nickname : defaultName;
    }

    public void setWaitingForNickname(Player player, boolean b) {
        waitingForNicknameInput.put(player.getUniqueId(), b);
    }

    public boolean isWaitingForNickname(Player player) {
        return waitingForNicknameInput.getOrDefault(player.getUniqueId(), false);
    }
}

File: src/main/java/org/hotamachisubaru/miniutility/Nickname/NicknameConfig.kt
package org.hotamachisubaru.miniutility.Nickname

import org.bukkit.configuration.file.FileConfiguration
import org.bukkit.configuration.file.YamlConfiguration
import org.bukkit.plugin.java.JavaPlugin
import java.io.File
import java.io.IOException
import java.util.*


class NicknameConfig(plugin: JavaPlugin) {
    private val config: FileConfiguration
    private val configFile = File(plugin.dataFolder, "nickname.yml")

    init {
        if (!configFile.exists()) {
            plugin.saveResource("nickname.yml", false)
        }
        this.config = YamlConfiguration.loadConfiguration(configFile)
    }

    fun getNickname(uuid: UUID): String {
        return config.getString("nicknames.$uuid", null)!!
    }

    fun setNickname(uuid: UUID, nickname: String?) {
        config["nicknames.$uuid"] = nickname
        saveConfig()
    }

    private fun saveConfig() {
        try {
            config.save(configFile)
        } catch (e: IOException) {
           error("設定ファイルの保存に失敗しました。")
        }
    }
}


